src/effects/native/loudnesscontoureffect.cpp

```diff
@@ -0,0 +1,159 @@
+}
+
+// static
+EffectManifest LoudnessContourEffect::getManifest() {
+    EffectManifest manifest;
+    manifest.setId(getId());
+    manifest.setName(QObject::tr("Loudness Contour"));
+    manifest.setShortName(QObject::tr("Loudness"));
+    manifest.setAuthor("The Mixxx Team");
+    manifest.setVersion("1.0");
+    manifest.setDescription(QObject::tr(
+        "A filter to compensate the loudness contour at lower volume"));
+    manifest.setEffectRampsFromDry(true);
+    manifest.setIsMixingEQ(true);
+
+    EffectManifestParameter* low = manifest.addParameter();
+    low->setId("loudness");
+    low->setName(QObject::tr("Loudness"));
+    low->setDescription(QObject::tr("Set the gain of the applied loudness contour"));
+    low->setControlHint(EffectManifestParameter::CONTROL_KNOB_LINEAR);
+    low->setSemanticHint(EffectManifestParameter::SEMANTIC_UNKNOWN);
+    low->setUnitsHint(EffectManifestParameter::UNITS_UNKNOWN);
+    low->setNeutralPointOnScale(1);
+    low->setDefault(-kMaxLoGain / 2);
+    low->setMinimum(-kMaxLoGain);
+    low->setMaximum(0);
+
+    EffectManifestParameter* killLow = manifest.addParameter();
+    killLow->setId("useMasterGain");
+    killLow->setName(QObject::tr("Master Gain"));
+    killLow->setDescription(QObject::tr("Follow master Gain"));
+    killLow->setControlHint(EffectManifestParameter::CONTROL_TOGGLE_STEPPING);
+    killLow->setSemanticHint(EffectManifestParameter::SEMANTIC_UNKNOWN);
+    killLow->setUnitsHint(EffectManifestParameter::UNITS_UNKNOWN);
+    killLow->setDefault(0);
+    killLow->setMinimum(0);
+    killLow->setMaximum(1);
+
+    return manifest;
+}
+
+LoudnessContourEffectGroupState::LoudnessContourEffectGroupState()
+        : m_oldMasterGain(1.0),
+          m_oldLoudness(0.0),
+          m_oldGain(1.0),
+          m_oldFilterGainDb(0),
+          m_oldUseMaster(false),
+          m_oldSampleRate(kStartupSamplerate) {
+
+    m_pBuf = SampleUtil::alloc(MAX_BUFFER_LEN);
+
+    // Initialize the filters with default parameters
+    m_low = std::make_unique<EngineFilterBiquad1Peaking>(kStartupSamplerate , kLoPeakFreq , kHiShelveQ);
+    m_high = std::make_unique<EngineFilterBiquad1HighShelving>(kStartupSamplerate , kHiShelveFreq ,kHiShelveQ);
+}
+
+LoudnessContourEffectGroupState::~LoudnessContourEffectGroupState() {
+    SampleUtil::free(m_pBuf);
+}
+
+void LoudnessContourEffectGroupState::setFilters(int sampleRate, double gain) {
+    m_low->setFrequencyCorners(
+            sampleRate, kLoPeakFreq, kLoPleakQ, gain);
+    m_high->setFrequencyCorners(
+            sampleRate, kHiShelveFreq, kHiShelveQ, gain / kHiShelveGainDiv);
+
+}
+
+LoudnessContourEffect::LoudnessContourEffect(
+                EngineEffect* pEffect, const EffectManifest& manifest)
+        : m_pLoudness(pEffect->getParameterById("loudness")),
+          m_pUseMasterGain(pEffect->getParameterById("useMasterGain")) {
+    Q_UNUSED(manifest);
+    m_pMasterGain = std::make_unique<ControlProxy>("[Master]", "gain");
```

Comment:
The user can always use the "Loudness" parameter. The master gain knob mapping works only in the use-case the sound system is leveled for full volume and the user used the master gain to reduce the loudness.  For all other use-cases this mapping is IMHO pointless.
But If we have other use cases, it would be no issue to to add them.

I have also considered to add it to the EQ Rack, but currently we allow only one master EQ. 
Or we can introduce a master Quick effect rack for it.